### PR TITLE
3992-wallet-storybook

### PR DESCRIPTION
**Pull request by DmitriyNoa**
This PR adds the storybook configuration. The CI pipeline will be added in a separate PR.

Screenshots:
In the WEB package, we are using Storybook storycap to take s screenshots, unfortunately, this is only web-add-on and it is not available on React native. 
As an alternative solution, we can use separate package for taking screenshots on React Native https://github.com/gre/react-native-view-shot and implement our own custom interface to access it. Screenshots should probably be in a different PR as we need to come up with an API.

### CODE REVIEW

File: packages/wallet/storybook/index.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { AppRegistry } from 'react-native';
+import { getStorybookUI, configure } from '@storybook/react-native';
+
+import './rn-addons';
+
+// import stories
+configure(() => {
+  require('./stories');
```
Comment (DmitriyNoa) [inline packages/wallet/storybook/index.tsx]: As per official Storybook documentation React native doesn't support dynamic story loading (https://storybook.js.org/docs/basics/writing-stories/#loading-stories). 

We have sevaral opitons:

- use react-native-storybook-loader to generate stories list via cli. The prestorybook script will generate the storyLoader file every time before the storybook is launched.

- Store all the stories in a single directory.
e.g.
```
configure(() => {
  require('../App/Components/Stories')
}, module)
```

- Just import all the stories manually. 
Stories.tsx
```
import './componentOne/componentOne.story'
import './componentTwo/componentTwo.story'
import './componentN/componentN.story'
```

I prefer the last one as it the most straight forward and fully compatible with TS

Comment (desfero) [reply packages/wallet/storybook/index.tsx]: ok, then let's go with the last one seems the easiest one for now.

File: packages/wallet/package.json [hunk 2]
```diff
@@ -28,14 +33,27 @@
...
     "@types/react-native": "^0.60.25",
     "@types/react-test-renderer": "16.9.1",
+    "@types/storybook__react": "^5.2.1",
     "@typescript-eslint/eslint-plugin": "^2.12.0",
     "@typescript-eslint/parser": "^2.12.0",
+    "babel-loader": "^8.0.6",
+    "babel-preset-react-native": "^4.0.1",
     "eslint": "^6.5.1",
-    "metro-react-native-babel-preset": "^0.56.0",
+    "metro-react-native-babel-preset": "^0.58.0",
```
Comment (desfero) [inline packages/wallet/package.json]: is there a need to update `metro-react-native-babel-preset`, unless really required maybe we can keep it in sync with `react-native` updates?

File: packages/wallet/package.json [hunk 3]
```diff
@@ -45,7 +63,15 @@
       "**/react-native/**",
       "**/react-native-gesture-handler",
       "**/react-native-gesture-handler/**",
-      "**/@react-navigation/**"
+      "**/@react-native-community/async-storage/**",
```
Comment (desfero) [inline packages/wallet/package.json]: fortunately we will get rid of all no hoists after we merge #4041

File: packages/wallet/storybook/addons.ts [hunk 4]
```diff
@@ -0,0 +1,2 @@
+import '@storybook/addon-actions/register';
```
Comment (desfero) [inline packages/wallet/storybook/addons.ts]: actions should be probably import from `addon-ondevice-actions` and moved t o`rn-addons` as they should be visible on device in the addons panel

File: packages/wallet/storybook/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+## Web integration
+React native storybook has an optional way to navigate storybook on the emulators or devices by running a web server with websocket connection to emulators.
+![React native storybook server](./assets/web-integration-demo.gif)
+
+Disclaimer: Currently there is a small issue with one of the packages  https://github.com/storybookjs/react-native/issues/13, when it will be solved we will be able to use web integration.
+
+Running storybook with web integration:
+- ```yarn storybook```
+- ```yarn storybook:ios``` or ```yarn storybook:android```
+Open http://localhost:7007 if it didn't open browser automatically.ya
```
Comment (desfero) [inline packages/wallet/storybook/README.md]: Seems like a typo `automatically.ya`?

File: packages/wallet/package.json [hunk 6]
```diff
@@ -28,14 +33,27 @@
...
+    "@storybook/addon-links": "^5.3.13",
+    "@storybook/addon-notes": "^5.3.13",
+    "@storybook/addon-ondevice-knobs": "^5.3.13",
+    "@storybook/addon-ondevice-notes": "^5.3.13",
+    "@storybook/addons": "5.3.13",
+    "@storybook/react-native": "^5.3.13",
+    "@storybook/react-native-server": "^5.3.13",
     "@types/react-native": "^0.60.25",
     "@types/react-test-renderer": "16.9.1",
+    "@types/storybook__react": "^5.2.1",
```
Comment (desfero) [inline packages/wallet/package.json]: looks like a not needed package?

File: packages/wallet/package.json [hunk 7]
```diff
@@ -28,14 +33,27 @@
     "@babel/core": "^7.6.2",
     "@babel/runtime": "^7.6.2",
     "@react-native-community/eslint-config": "^0.0.5",
+    "@storybook/addon-actions": "^5.3.13",
+    "@storybook/addon-knobs": "^5.3.13",
+    "@storybook/addon-links": "^5.3.13",
+    "@storybook/addon-notes": "^5.3.13",
+    "@storybook/addon-ondevice-knobs": "^5.3.13",
+    "@storybook/addon-ondevice-notes": "^5.3.13",
+    "@storybook/addons": "5.3.13",
```
Comment (desfero) [inline packages/wallet/package.json]: looks like a not used package?

File: packages/wallet/package.json [hunk 8]
```diff
@@ -7,17 +7,22 @@
...
+    "storybook:android": "ENVFILE=./storybook/.storybook-config react-native run-android",
     "start:devtools": "react-devtools",
     "lint": "eslint . --ext .js,.jsx,.ts,.tsx",
-    "prepare": "echo 'Nothing to prepare for wallet yet...'"
+    "prepare": "echo 'Nothing to prepare for wallet yet...'",
+    "storybook": "(adb reverse tcp:7007 tcp:7007 || true) && start-storybook"
   },
   "dependencies": {
     "@neufund/shared": "^1.0.0",
+    "@react-native-community/async-storage": "^1.7.1",
```
Comment (desfero) [inline packages/wallet/package.json]: Seems not used at the moment?

File: packages/wallet/index.ts [hunk 9]
```diff
@@ -1,7 +1,9 @@
 import "react-native-gesture-handler";
-
+import Config from "react-native-config";
 import { AppRegistry } from "react-native";
 import { App } from "./app/App";
 import { name as appName } from "./app.json";
+import StorybookUIRoot from "./storybook"
+const EntryPoint = Config.STORYBOOK_RUN == 1 ? StorybookUIRoot : App;
```
Comment (desfero) [inline packages/wallet/index.ts]: For consistency let's use `Config.STORYBOOK_RUN === "1"`

File: packages/wallet/app/components/Button.tsx [hunk 10]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { Button } from "react-native";
+
+interface ButtonProps {
```
[the remark this answers is not in the document]
Comment (DmitriyNoa) [reply packages/wallet/app/components/Button.tsx]: Yes, this is a good approach, I didn't intend to actually make this component and leave it. It was just an example. But I'll add it and maybe move the Button to a common folder incode components one.

File: packages/wallet/app/components/Button.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,15 @@
+import * as React from "react";
+import { storiesOf } from "@storybook/react-native";
+import {View} from "react-native";
+import {NeuButton} from "./Button";
+
+storiesOf('NeuButton', module)
+  .addParameters({
+    screenshot: {
```
[the remark this answers is not in the document]
Comment (DmitriyNoa) [reply packages/wallet/app/components/Button.stories.tsx]: Yes, this is a leftover from testing. I'll remove it

File: packages/wallet/types/react-native-dotenv.d.ts [hunk 12]
```diff
@@ -0,0 +1 @@
+declare module "react-native-dotenv";
```
[the remark this answers is not in the document]
Comment (DmitriyNoa) [reply packages/wallet/types/react-native-dotenv.d.ts]: Removed

File: packages/wallet/app/components/common/buttons/Button.tsx [hunk 13]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { Button as ReactNativeButton } from "react-native";
+import { TDataTestId } from "@neufund/web/app/types";
```
Comment (desfero) [inline packages/wallet/app/components/common/buttons/Button.tsx]: We should not use `TDataTestId` as it's specific for the web only in RN there is already build in `testId` prop

File: packages/wallet/app/components/common/buttons/Button.tsx [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+
+type TExternalProps = {
+  title: string;
+  onPress: (evt: any) => void;
+};
+type TButtonProps = React.ComponentProps<typeof ReactNativeButton>;
+
+const Button: React.FunctionComponent<TExternalProps & TButtonProps & TDataTestId> = props => {
+  return <ReactNativeButton title={props.title} onPress={props.onPress} />;
+};
```
Comment (desfero) [inline packages/wallet/app/components/common/buttons/Button.tsx]: Seems that we can a little bit simplify the code to just spread the props.
```
const Button: React.FunctionComponent<TButtonProps> = props => {
  return <ReactNativeButton {...props} />;
};
```

**Pull request merged**